client/src/utils/userdata.ts

```diff
@@ -1,4 +1,44 @@
 // Functions to retrieve and manipulate data in local storage
 
-export const getUser: () => { hash?: string; username?: string } = () =>
-  JSON.parse(localStorage.getItem("user") || "{}");
+import type { Chats, Message, User } from "../type/global";
+
+enum Keys {
```

Comment:
It would be better to name enums as singular nouns.
`Key.Chat` is the chat key.
Where keys makes more sense for `const keys = [...]` a collection of keys.

Purely a naming preference not a required change but is what seems to be the more convention and what I would recommend.

Reply:
I applied this change!